### PR TITLE
update module system docs

### PR DESCRIPTION
**Pull request by ublubu**
@wlan0

### CODE REVIEW

File: docs/modules/index.md [hunk 1]
```diff
@@ -2,12 +2,26 @@
...
+It _imports_ configuration values from a module called `env.short.yaml`
+  and accepts a _parameter_ to configure its `name`.
+
 
 ```yaml
 imports:
 - env: ./env.short.yaml
 params:
-- name: for selector/labels and name
+- name: "used for selector/labels and name"
```
Comment (wlan0) [inline docs/modules/index.md]: Add yaml comments saying what this field represents 

```
name: "used for selector/labels and name" # Description of the param
```

**Comment by wlan0** [comment]
I just read the beginning of this file. The beginning should include more contextual information about how users should think about the module system following this structure. The docs are the best method we have to communicate to the user. 

1. Why are modules needed
2. How they are designed
3. How the user is expected to use them
4. Specifics of the features this module system provides

Multiple files may be required here.

**Comment by ublubu** [comment]
updated for points 1, 3, and (in combination with existing material) 4. @wlan0

**Comment by ublubu** [comment]
updated, @wlan0

**Pull request merged**